crates/db_schema/src/impls/instance.rs

```diff
@@ -94,6 +96,37 @@ impl Instance {
       .await
   }
 
+  /// returns a list of all instances, each with a flag of whether the instance is allowed or not
+  /// ordered by id
+  pub async fn read_all_with_blocked(pool: &mut DbPool<'_>) -> Result<Vec<(Self, bool)>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let use_allowlist = federation_allowlist::table
```

Comment:
This check is probably unecessary, and also adds an extra query that has to be waited on before the next one can start. Likely much faster to just always join to both.

Reply:
it is not possible to do this in one query because the meaning of the entries in the blocklist depends on the existence of any row at all in the allow list